tests/admin_views/test_adminsite.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import datetime
+
+from django.conf.urls import include, url
+from django.contrib import admin
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.test import TestCase, override_settings
+from django.test.client import RequestFactory
+
+from .models import Article
+
+site6 = admin.AdminSite(name="admin6")
+site6.register(User)
+site6.register(Article)
+
+urlpatterns = [
+    url(r'^test_admin/admin6/', include(site6.urls)),
+]
+
+
+@override_settings(
+    PASSWORD_HASHERS=['django.contrib.auth.hashers.SHA1PasswordHasher'],
+    ROOT_URLCONF="admin_views.test_adminsite",
+)
+class SiteEachContextTest(TestCase):
+    """
+    Check each_context contains the documented variables, check that available_apps context
+    variable structure is the expected one
+    """
+    @classmethod
+    def setUpTestData(cls):
+        cls.u1 = User.objects.create(
+            id=100, password='sha1$995a3$6011485ea3834267d719b4c801409b8b1ddd0158',
+            last_login=datetime.datetime(2007, 5, 30, 13, 20, 10), is_superuser=True, username='super',
+            first_name='Super', last_name='User', email='super@example.com',
+            is_staff=True, is_active=True, date_joined=datetime.datetime(2007, 5, 30, 13, 20, 10)
```

Comment:
prefer including a trailing comma in kwargs so if more items are added in the future we don't have to modify this line again